active-record/src/main/java/com/iluwatar/active/record/MagicWand.java

```diff
@@ -0,0 +1,264 @@
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static MagicWand find(long id) {
+
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(SELECT_SQL);
+
+            ResultSet rs;
+            ps.setLong(1, id);
+            rs = ps.executeQuery();
+            if (rs.next()) {
+                final MagicWand wand = new MagicWand();
+
+                wand.id = rs.getLong("id");
+                wand.setCore(WandCoreType.valueOf(rs.getString("core")));
+                wand.setWood(WandWoodType.valueOf(rs.getString("wood")));
+                wand.setLengthInches(rs.getDouble("length_inches"));
+
+                return wand;
+            }
+            else {
+                return null;
+            }
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    private void validateToSave() {
+        validateFieldsToBeNotNull();
+        if (this.getId() != null)
+            throw new IllegalStateException("Can not save wand that was previously saved. Use 'update' metod instead.");
+    }
+
+    private void validateToUpdate() {
+        validateFieldsToBeNotNull();
+        if (this.getId() == null) throw new IllegalStateException("Can not update a record without ID specified");
+    }
+
+    private void validateToDelete() {
+        if (this.getId() == null) throw new IllegalStateException("Can not delete a record without ID specified");
+    }
+
+    private void validateFieldsToBeNotNull() {
+        if (this.getLengthInches() == null)
+            throw new IllegalStateException("Can not save a wand without length specified");
+        if (this.getWood() == null)
+            throw new IllegalStateException("Can not save a wand without wood specified");
+        if (this.getCore() == null)
+            throw new IllegalStateException("Can not save a wand without core specified");
+    }
+
+    /**
+     * ***********************************
+     * ****** Domain Logic
+     * ***********************************
+     */
+
+    private double calculateWoodMagicFactor() {
+
+        switch (this.getWood()) {
+            case HOLLY:
+                return 1.2d;
+            case WINE:
+                return 1.0d;
+            case HAWTHORN:
+                return 0.8d;
+            default:
+                return 0.0d;
+        }
+    }
+
+    private double calculateCoreMagicFactor() {
+
+        switch (this.getCore()) {
+            case PHOENIX_FEATHER:
+                return 1.2d;
+            case DRAGON_HEARTSTRING:
+                return 1.0d;
+            case UNICORN_HAIR:
+                return 0.8d;
+            default:
+                return 0.0d;
+        }
+    }
+
+    public double calculateMagicPower() {
+
+        return calculateWoodMagicFactor() * calculateWoodMagicFactor() * this.getLengthInches();
```

Comment:
You should be consistent with either using `this` for instance methods or not at all. I prefer not using them if not required.